### PR TITLE
Dsm switcher button

### PR DESCRIPTION
**Pull request by pierallard**
https://www.notion.so/akeneo/Switcher-button-chipmunks-220e98b2359a4d209f8d4135d2873e3b

### CODE REVIEW

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+<ArgsTable story="Standard" />
+
+## Variation on inline
+
+<Canvas>
+  <Story name="Inline">
+    {args => {
+      return (
+        <>
+          <SwitcherButton {...args} label={'Inline'} inline={true}>
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx:46]:
```suggestion
          <SwitcherButton label="Inline" inline>
```

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            True
+          </SwitcherButton>
+          <SwitcherButton {...args} label={'Inline'} inline={false}>
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx:49]:
```suggestion
          <SwitcherButton label="Inline">
```

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+## Variation on deletable
+
+<Canvas>
+  <Story name="Deletable">
+    {args => {
+      return (
+        <>
+          <SpaceContainer width={200}>
+            <SwitcherButton {...args} label={'Deletable'} inline={false} deletable={true}>
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx:66]:
```suggestion
            <SwitcherButton label="Deletable" deletable>
```

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+              True
+            </SwitcherButton>
+          </SpaceContainer>
+          <SpaceContainer width={200}>
+            <SwitcherButton {...args} label={'Deletable'} inline={false} deletable={false}>
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx:71]:
```suggestion
            <SwitcherButton label="Not deletable">
```

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+</Canvas>
+
+## Example with flag
+
+<Canvas>
+  <Story name="Flag">
+    {args => {
+      return (
+        <>
+          <SwitcherButton {...args} label={'Locale'} inline={true}>
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx:88]:
```suggestion
          <SwitcherButton label="Locale" inline>
```

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            <Locale code='en_US' languageLabel='English'/>
+          </SwitcherButton>
+          <SwitcherButton {...args} label={'Locale'} inline={false}>
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.stories.mdx:91]:
```suggestion
          <SwitcherButton label="Locale" inline>
```

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+  children?: ReactNode;
+};
+
+/**
+ * TODO.
+ */
+const SwitcherButton = React.forwardRef<HTMLDivElement, SwitcherButtonProps>(
+  (
+    {label, children, onClick, deletable = false, onDelete, inline = true, ...rest}: SwitcherButtonProps,
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.tsx:110]: What do you think about a default value at false (maybe change inline to multiline if you prefer having multiline as default)? 

```suggestion
    {label, children, onClick, deletable = false, onDelete, inline = false, ...rest}: SwitcherButtonProps,
```

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.tsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    forwardedRef: Ref<HTMLDivElement>
+  ) => {
+    const handleDelete = () => {
+      if (deletable && onDelete) {
+        onDelete();
+      }
+    };
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.tsx:117]:
```suggestion
    const handleDelete = () => deletable && onDelete.?();
```

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.tsx [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+    const handleClick = () => {
+      if (onClick) {
+        onClick();
+      }
+    };
```
Comment (juliensnz) [inline akeneo-design-system/src/components/SwitcherButton/SwitcherButton.tsx:123]:
```suggestion
    const handleClick = () => onClick.?();
```

**Review by juliensnz — APPROVED** [review]
Thanks a lot for this PR! GTM after fix

**Review by olivierpontier — APPROVED** [review]

File: akeneo-design-system/src/components/SwitcherButton/SwitcherButton.tsx (hunk 7 above)
Comment (pierallard) [reply akeneo-design-system/src/components/SwitcherButton/SwitcherButton.tsx:110]: As the DSM default is inline, I think the default value should be true.

**Pull request merged**